### PR TITLE
Adds ADR for historical analytics approach

### PR DESCRIPTION
**Pull request by JPrevost**
Why are these changes being introduced:

* There are multiple ways we can address this problem
* Using an ADR to provide context and options, gathering feedback on the proposed solution, and then merging the decision allows us to see what else we considered in addition to what we eventually implement.

Relevant ticket(s):

https://mitlibraries.atlassian.net/browse/TCO-16

### CODE REVIEW

**Review by matt-bernhardt — APPROVED** [review]
I'm going to tentatively approve this ADR, subject to some of the questions I raise above. My strongest question is whether you're proposing these exact columns (including journal_title and staff_categorized), or if those are only illustrative.

My other comment is me raising some alternative structures, and working through why I think they don't actually work. I'm not sure whether this means that we should list them in the ADR text, but - unless you see something in here that I don't - I don't think they change the decision you're proposing.

File: docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    TotalMatchesByAlgorithm: +Integer id
+    TotalMatchesByAlgorithm: +Date month 
+    TotalMatchesByAlgorithm: +Integer doi
+    TotalMatchesByAlgorithm: +Integer issn
+    TotalMatchesByAlgorithm: +Integer isbn
+    TotalMatchesByAlgorithm: +Integer pmid
+    TotalMatchesByAlgorithm: +Integer journal_title
+    TotalMatchesByAlgorithm: +Integer staff_categorized
+    TotalMatchesByAlgorithm: +Integer no_matches
+```
```
[the remark this answers is not in the document]
Comment (jazairi) [reply docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md:45]: Regarding question 2, my preference would be to add those columns when we're ready to use them. However, I like including them in the class diagram as an indication of future state.

File: docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+We already have a record of every time we’ve seen a given search for any phrase - adding a table (or even fields in the SearchEvent table) for each detector we build would allow us to have a record of whether each detector fired, and what it returned. Setting a default value of NULL on each field, but storing an empty non-NULL value via code, would allow us to query for failures to detect (which would be stored as {} separately from any search prior to that detector being in place (which would be stored as NULL values).
+
+An approach like this, which doesn’t generate only counts but also the details of what is detected / returned, would provide greater visibility into application performance - but at the cost of needing to be recorded at query time, and not as a batch retroactively.
+
+This stores a lot more data and it isn't clear what we would use it for. This approach should likely be implemented in the future if we find we have a specific question that can only be solved with this level of detail. We would lose historical data if we wait to implement this, but that risk is outweighed by avoiding storing a bunch of data without a clear purpose.
```
Comment (jazairi) [inline docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md:119]: Alternatively, we could store the data for now and drop the table if it becomes clear that there's no purpose for it. The downside of that (aside from storing a lot of data in the interim) is knowing _when_ to decide whether it's useful or not.

**Review by jazairi — APPROVED** [review]
I agree with the decision here. The more verbose detections table is intriguing, but it seems to me that the extra data would not serve the intended purpose here of validating how often our algorithms are finding matches.

File: docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md (hunk 2 above)
Comment (matt-bernhardt) [reply docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md:119]: I concur - knowing when to get rid of that data feels like a thornier process than just not collecting it yet until we have a more clear understanding of how we'll handle it and when to delete it.

File: docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JPrevost) [reply docs/architecture-decisions/0005-use-multiple-minimal-historical-analytics-models.md:45]: 1. yes
2. yes, we'd only add fields as we need them not predict what we might need later
3. good point. Adding a "if this PR adds a new algorithm please make sure analytics account for it" kinda thing makes sense.

**Comment by JPrevost** [comment]
Noting that we discussed in zoom and decided to merge this PR as-is with the understanding that the database structures here were examples and we'll confirm during PRs the exact structures that will be used.

**Pull request merged**